### PR TITLE
amr-to-template first pass

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
Add in function to break an AMR into smaller AMRs in the form of MIRA-template models.  This is plumbing work, the script amr_to_templates.py is provisionally working, but will likely need to be modified to add in the notion of junction points and bug fixes. CC @liunelson 

For @shawnyama the usage in Terarium will look something like this:
```
const context = {
  context: 'mira_model',
  language: 'python3',
  context_info: {
    id: '8f1b00a3-00bd-47b1-925a-3e1227a0af9a'
  }
};

const manager = new KernelSessionManager();
await manager.init('beaker_kernel', 'Beaker Kernel', context);

manager
  .sendMessage('amr_to_templates', {})
  .on('amr_to_templates_response', (d) => {
    blah.value = d.content.templates;
  });
```

### CODE REVIEW

File: src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# Define a new MIRA TemplateModel for each template in the model
+templates_tm = []
+templates_json = []
+for i, __ in enumerate(model_tm.templates):
+    t = copy.deepcopy(model_tm.templates[i])
+    tm = TemplateModel(
+        templates = [t],
+        # parameters = {params_global[p]: params_concepts_global[params_global[p]] for p in t.get_parameter_names()},
+        # initials = {vars_global[c.name]: inits_concepts_global[c.name] for c in t.get_concepts()},
+        # annotations = Annotations(name = f"{t.name}{i}")
```
Comment (YohannParis) [inline src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py:18]: Why are those kept?

File: src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        parameters = {p: model_tm.parameters[p] for p in t.get_parameter_names()},
+        initials = {v: model_tm.initials[v] for v in t.get_concept_names()},
+        annotations = Annotations(name = f"{t.name}"),
```
Comment (YohannParis) [inline src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py:21]: Why do not need the index `i` here, compare to the commented out line?

File: src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py (hunk 1 above)
Comment (mwdchang) [reply src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py:18]: Algorithm is not one, just enough here to get people unblocked.

File: src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py (hunk 2 above)
Comment (mwdchang) [reply src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py:21]: This is @liunelson code, will let Nelson comment.

**Pull request merged**